packages/trader/src/App/Components/Elements/ContractAudit/contract-audit.jsx

```diff
@@ -6,70 +6,50 @@ import { WS } from 'Services/ws-methods';
 import ContractDetails from './contract-details.jsx';
 import ContractHistory from './contract-history.jsx';
 
-class ContractAudit extends React.Component {
-    state = {
-        contract_update_history: [],
-    };
+const ContractAudit = ({ has_result, toggleHistoryTab, is_multiplier, contract_update_history, ...props }) => {
+    const { contract_id, currency } = props.contract_info;
+    const [update_history, setUpdateHistory] = React.useState([]);
 
-    static getDerivedStateFromProps({ is_multiplier, contract_update_history }, state) {
-        if (!is_multiplier) return state;
-        if (!!contract_update_history.length && contract_update_history.length > state.contract_update_history.length) {
-            return {
-                contract_update_history,
-            };
-        }
-        return state;
-    }
+    React.useEffect(() => {
+        if (!!contract_update_history.length && contract_update_history.length > update_history.length)
+            setUpdateHistory(contract_update_history);
+    }, [contract_update_history]);
```

Comment:
Don't we need `update_history` in deps?

Reply:
@salar-fs  We had `static getDerivedStateFromProps` which was returning an object to update the state. Now I'm replacing it with the above `useEffect`, I want to update `update_history` with `contract_update_history` in some conditions when `contract_update_history` change.
Base on the previous implementation on other cases the return amount was `state`, I think no need to do `setUpdateHistory` for `update_history` itself, am I right?

Reply:
Fine. Any `eslint` warnings then? Did you check?

Reply:
I see what you mean, let me add it in deps.